rrule/src/iter/utils.rs

```diff
@@ -1,33 +1,44 @@
 use crate::core::DateTime;
 use chrono::{TimeZone, Utc};
 use chrono_tz::UTC;
+use lazy_static::lazy_static;
+use std::collections::HashMap;
+use std::sync::Mutex;
+
+const DAY_SECS: i64 = 24 * 60 * 60;
+
+lazy_static! {
+    static ref LEAP_YEAR_CACHE: Mutex<HashMap<i32, bool>> = Mutex::new(HashMap::new());
```

Comment:
... and this

Reply:
I am not so sure about this. Seems like introducing a `Mutex` for a quick and simple calculation is unnecessary. This can also have unintentional consequences with lock contention, whereas the simple calculation can be done 100% in parallel between different threads.

Reply:
Or we can use one of the existing libraries, like https://crates.io/crates/lru or https://github.com/jaemk/cached or [...](https://crates.io/keywords/cache)

Reply:
I just benchmarked again, the gain of this is not that much (I remembered something around 20% improvement)
So I'll revert it back 👍🏼

Reply:
It is actually quite surprising to me that locking + hashing the year can be faster than the existing code, maybe hashing is even faster than I thought. If it turns out that this part of the code shows up in the flamegraph again, then I think we can prebuild a leap year cache at compile time from year 1900 to 2100 or something like that, and only compute it if year is outside that which should be rare.